src/conn/conn_sweep.c

```diff
@@ -45,6 +47,31 @@ __sweep_mark(WT_SESSION_IMPL *session, uint64_t now)
           __wt_atomic_loadi32(&dhandle->session_inuse) > 0 || dhandle->timeofdeath != 0)
             continue;
 
+        /*
+         * When MongoDB opens cursors on table URIs, WiredTiger opens and releases table dhandles,
+         * particularly for simple tables. This repetitive dhandle creation and release, handled by
+         * the sweep server, leads to schema lock contention. To prevent this, the below algorithm
+         * modifies the sweep server's behavior. The algorithm now checks if any file dhandles are
+         * associated with the table before marking a table dhandle for sweeping, thus preventing
+         * unnecessary closures.
+         */
+        if (dhandle->type == WT_DHANDLE_TYPE_TABLE) {
+            table = (WT_TABLE *)dhandle;
+            if (table->is_simple && table->cgroups != NULL) {
+                const char *uri = table->cgroups[0]->source;
+                WT_WITHOUT_DHANDLE(session,
+                  WT_WITH_HANDLE_LIST_READ_LOCK(
+                    session, (ret = __wt_conn_dhandle_find(session, uri, NULL))));
```

Comment:
can we add "it operates only on the dhandles that are already opened." comment here?